src/backend/backend.h

```diff
@@ -271,6 +271,11 @@ namespace snmalloc
       return pagemap.template get<potentially_out_of_range>(p);
     }
 
+    static MetaEntry& get_meta_data_mut(address_t p)
```

Comment:
I am not keen to expose this.  It is hard to implement in the sandbox RPC case. I would be much keener if we could use the metaslab for this.

Reply:
I agree - I missed the layering when I read the code.  In the sandbox case, the pagemap is read-only in the child and so anything that updates it must be part of the backend API, where they can become RPCs.  

I think, after looking at @ihaller's code, it would be good if we could avoid exposing `MetaEntry` outside of the back end entirely.  The back end is then free to use whatever storage it wants, including using a larger structure in the pagemap.